---
title: "Home"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is the page for the cancer-germline project. Below are links to analyses.

I ran S-LDSC on annotations from ENCODE, Roadmap, TCGA, and other papers.

* [Enrichment Analyses](s-ldsc_enrichment.html) (03/09/2020)

Annotations were merged into broad categories (immune histones, immune open chromatin, etc.). Torus was ran to estimate prior probabilities per SNP. SuSiE was ran with L = 1. SuSiE was also ran with no prior probability, and the resulting PIPs were compared below.

* [Effect of Torus Prior on Finemapping](annot_effect.html) (03/20/2020)

Torus + SuSiE finemapping results for immune SNPs with PIP > 0.05

* [Spreadsheets of finemapping results](https://docs.google.com/spreadsheets/d/1MfWZtwy5h9ubmL7qoNPNLbHCS15WlDXUeqvXZmzDK5U/edit?usp=sharing) (03/24/2020)

HiC data for GM12878 (LCLs) from Bing Ren et al. were obtained ( only the significant links were kept). These were merged with HiC data with CHiCAGO score > 5 for 17 immune primary cell types from Ulirsch, J.C., Lareau, C.A., Bao, E.L. et al. Interrogation of human hematopoiesis at single-cell and single-variant resolution. Nat Genet 51, 683â€“693 (2019).

Immune fine-mapped SNPs were matched to genes via the HiC data. A gene-level PIP is constructed at:

* [HiC Gene PIPs](https://docs.google.com/spreadsheets/d/1C8O2uFupAgsIYMElahWP0LLt4rJo4k2bdOfgT9htkGE/edit?usp=sharing) (03/24/2020)

The ones with obvious immune function are highlighted in red. 

